src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DescriptorMarker.cs

```diff
@@ -85,15 +86,17 @@ private void ProcessNamespaces(ModuleDesc assembly, XPathNavigator nav)
 
                 if (!foundMatch)
                 {
-                    // LogWarning(namespaceNav, DiagnosticId.XmlCouldNotFindAnyTypeInNamespace, fullname);
+#if !READYTORUN
+                    LogWarning(namespaceNav, DiagnosticId.XmlCouldNotFindAnyTypeInNamespace, fullname);
+#endif
                 }
             }
         }
 
         private void MarkAndPreserve(TypeDesc type, XPathNavigator nav, TypePreserve preserve)
         {
             var bindingOptions = preserve switch {
-                TypePreserve.Methods => DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods,
+                TypePreserve.Methods => DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods | DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors,
```

Comment:
Why we have to do this and illinker doesn't?

Reply:
We need to answer this before merging - otherwise I think this change looks good.

Reply:
I guess in part because illink descriptor implementation preceded DynamicallyAccessedMembers usage, so instead the descriptor in illink actually does not preserve any member, it just saves the info of what needs to be preserved for a later stage (MarkStep). Given that NativeAOT does not have global storage, nor uses the MarkStep, we needed to call the dependency graph and we already wrote all that logic in the Binder so I just reused it.

Reply:
I'm OK with using the binder (migth think about porting that to linker eventually as well). But please add a comment that linker does it differently right now... and how differently.